src/globals.ts

```diff
@@ -1,12 +1,123 @@
 import * as logger from './logger';
 import { Context } from 'aws-lambda';
-
 const MAX_TAGS = 50;
 const MAX_TAG_KEY_LEN = 50;
 const MAX_TAG_VALUE_LEN = 50;
 const ADD_TAG_ERROR_MSG_PREFIX = 'Skipping addExecutionTag: Unable to add tag';
 export const DEFAULT_MAX_SIZE_FOR_REQUEST = 1000 * 1000;
 
+export const Timer = (() => {
+  let time = 0;
+
+  type CockState = 'stopped' | 'idle' | 'counting';
+
+  let state: CockState = 'stopped';
+  let startTime = 0;
+  let _duration = 0;
+
+  // since utils uses globals had to duplicate function
+  const runOneTimeWrapper = (func: Function, context: any): Function => {
+    let done = false;
+    return (...args) => {
+      if (!done) {
+        const result = func.apply(context || this, args);
+        done = true;
+        return result;
+      }
+    };
+  };
+
+  const warnTimeout = runOneTimeWrapper(() => {
+    logger.warnClient('Lumigo tracer reached its timeout and will no longer collect data');
+  }, {});
+
+  const init = (duration) => {
+    time = 0;
+    startTime = 0;
+    state = 'idle';
+    _duration = duration;
+  };
+
+  const start = () => {
+    if (state === 'stopped') {
+      init(process.env.LUMIGO_TRACER_TIMEOUT || 500);
```

Comment:
Let's do also some calculation with the lambda timeout if it's 1s. let's define the max to be `timeout / 5 (200 ms)` max of `500ms`